dask/dataframe/backends.py

```diff
@@ -67,7 +67,15 @@ def make_meta_index(x, index=None):
     return x[0:0]
 
 
-meta_object_types = (pd.Series, pd.DataFrame, pd.Index, pd.MultiIndex)
+meta_object_types = (
+    pd.Series,
+    pd.DataFrame,
+    pd.Index,
+    pd.MultiIndex,
+    dict,
+    list,
+    tuple,
+)
```

Comment:
@GenevieveBuckley As part of fixing an issue with meta creation when user is operating in CPU & GPU boundaries: https://github.com/rapidsai/cudf/issues/7946, I fixed the issue with meta creation by introducing a new dispatch `make_meta_obj` to handle generic `object` types. But this has to be registered against a specific backend and not againt `object`. As a result we had to introduce a middle-man function `make_meta_util`, this is similar to how [concat](https://github.com/dask/dask/blob/ac1bd05cfd40207d68f6eb8603178d7ac0ded922/dask/dataframe/methods.py#L395) currently works as a middle-man function. 

Making this addition will have the issue re-surface, did you try using `make_meta_util` without making these changes?

Reply:
We use `dask.dataframe.utils.make_meta` for dicts in our code-base. I felt like this was public API, especially as it get's advertised in the docs, e.g. here:
https://github.com/dask/dask/blob/d2020bf30962c9660fba399392a3e8c9df59dca4/dask/dataframe/core.py#L638-L639
Our codebase is incompatible with `dask=2021.05.1`. Using `make_meta_util` would be an option, however requires us to use a `if LooseVersion(dask.__version__) >= "2021.05.1"` type switch in our code to support older versions of dask. I would prefer if this could be avoided, e.g. by renaming `make_meta_util` to `make_meta` as suggested by @GenevieveBuckley above.

Reply:
I agree with @GenevieveBuckley that we should document `make_meta_util`, but I feel renaming `make_meta_util` to `make_meta` will lead to confusion as there will then be two `make_meta`'s one is a `dispatch` and the other is a `method`. In which case we will need to rename the `make_meta`(dispatch) to `make_meta_dispatch`. 

To avoid renaming everything I think we should rather stop exposing `make_meta`(method) through docs and instead advertise `make_meta_util` with new behavior.

cc: @quasiben @jakirkham